### PR TITLE
Updating code to reflect API change

### PR DESCRIPTION
**Pull request by udesou**
Merge after https://github.com/mmtk/mmtk-core/pull/897 to reflect the API change.

### CODE REVIEW

File: mmtk/Cargo.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 name = "mmtk_openjdk"
 version = "0.19.0"
 authors = [" <>"]
-rust-version = "1.66.0"
+rust-version = "1.71.1"
```
Comment (qinsoon) [inline mmtk/Cargo.toml:5]: This is the minimal Rust version, not the version we use for building and running tests. MSRV should be 1.70, and the version we use to run tests should be 1.71.1 (in the `rust-toolchain` file).

Comment (udesou) [reply mmtk/Cargo.toml]: Gotcha. Fixed it!

**Comment by udesou** [comment]
@qinsoon Do you know why I'd be getting failing tests here?

**Comment by qinsoon** [comment]
> @qinsoon Do you know why I'd be getting failing tests here?

It should be the mark compact issue we talked about in the meeting. You can merge `master` to the mmtk-core PR to include [42991c2e43f251214a1a9be2c24933f9c2339f35](https://github.com/mmtk/mmtk-core/commit/42991c2e43f251214a1a9be2c24933f9c2339f35) in the PR, and the issue should be fixed.

**Review by qinsoon — APPROVED** [review]

**Pull request merged**